File: 프로그래머스/unrated/250135. ［PCCP 기출문제］ 3번 ／ 아날로그 시계/［PCCP 기출문제］ 3번 ／ 아날로그 시계.java
class Solution {
    public int solution(int h1, int m1, int s1, int h2, int m2, int s2) {
        int start = h1 * 3600 + m1 * 60 + s1;
        int end = h2 * 3600 + m2 * 60 + s2;
        
        int result = 0;
        if (start == 0 || start == 43200) {
            result++;
        }
        
        while (start < end) {
            
            double rad_h = (start / 120.0) % 360;
            double rad_m = (start / 10.0) % 360;
            double rad_s = (start * 6.0) % 360;
            
            int next = start + 1;
            double next_rad_h = (next / 120.0) % 360;
            double next_rad_m = (next / 10.0) % 360;
            double next_rad_s = (next * 6.0) % 360;
            
            next_rad_h = next_rad_h == 0 ? 360 : next_rad_h;
            next_rad_m = next_rad_m == 0 ? 360 : next_rad_m;
            next_rad_s = next_rad_s == 0 ? 360 : next_rad_s;
            
            if (next_rad_h == next_rad_m && next_rad_m == next_rad_s) {
                result++;
                start++;
                continue;
            }
            
            if (rad_s < rad_h && next_rad_h <= next_rad_s) {
                result++;
            }
            
            if (rad_s < rad_m && next_rad_m <= next_rad_s) {
                result++;
            }
            
            start++;
        }
        
        return result;
    }
}

File: 프로그래머스/unrated/250135. ［PCCP 기출문제］ 3번 ／ 아날로그 시계/［PCCP 기출문제］ 3번 ／ 아날로그 시계.py
def solution(h1, m1, s1, h2, m2, s2):
    start = h1 * 3600 + m1 * 60 + s1
    end = h2 * 3600 + m2 * 60 + s2
    
    cnt = 0
    
    if start == 0 or start == 43200:
        cnt += 1
    
    # 시침 : 1/120 * (t % 43200)
    # 분 : 1/10 * (t % 3600)
    # 초 : 6 * (t % 60)
    
    while start < end:
        rad_h =  1/120 * (start % 43200)
        rad_m = 1/10 * (start % 3600)
        rad_s = 6 * (start % 60)
        
        next = start + 1
        next_rad_h =  1/120 * (next % 43200)
        next_rad_m = 1/10 * (next % 3600)
        next_rad_s = 6 * (next % 60)
      
        if next_rad_h == 0:
            next_rad_h = 360
        if next_rad_m == 0:
            next_rad_m = 360
        if next_rad_s == 0:
            next_rad_s = 360
        
        if next_rad_h == next_rad_m == next_rad_s:
            cnt += 1
            start +=1
            continue
            
        
        if rad_h > rad_s and next_rad_h <= next_rad_s:
            cnt += 1
        if rad_m > rad_s and next_rad_m <= next_rad_s:
            cnt += 1
            
        start += 1
    
    
    
    return cnt